pinot-core/src/test/java/org/apache/pinot/core/data/readers/RecordReaderSampleDataTest.java

```diff
@@ -0,0 +1,229 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.readers;
+
+import com.google.common.base.Preconditions;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import org.apache.pinot.common.data.FieldSpec;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.core.data.GenericRow;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+
+public class RecordReaderSampleDataTest {
+  private static final File AVRO_SAMPLE_DATA_FILE = new File(Preconditions
+      .checkNotNull(RecordReaderSampleDataTest.class.getClassLoader().getResource("data/test_sample_data.avro"))
+      .getFile());
+  private static final File CSV_SAMPLE_DATA_FILE = new File(Preconditions
+      .checkNotNull(RecordReaderSampleDataTest.class.getClassLoader().getResource("data/test_sample_data.csv"))
+      .getFile());
+  private static final File JSON_SAMPLE_DATA_FILE = new File(Preconditions
+      .checkNotNull(RecordReaderSampleDataTest.class.getClassLoader().getResource("data/test_sample_data.json"))
+      .getFile());
+  private final Schema SCHEMA = new Schema.SchemaBuilder().addSingleValueDimension("column1", FieldSpec.DataType.LONG)
+      .addSingleValueDimension("column2", FieldSpec.DataType.INT)
+      .addSingleValueDimension("column3", FieldSpec.DataType.STRING)
+      .addSingleValueDimension("column7", FieldSpec.DataType.STRING)
+      .addSingleValueDimension("unknown_dimension", FieldSpec.DataType.STRING)
+      .addMetric("met_impressionCount", FieldSpec.DataType.LONG).addMetric("unknown_metric", FieldSpec.DataType.DOUBLE)
+      .build();
+  private final Schema SCHEMA_SAME_INCOMING_OUTGOING = new Schema.SchemaBuilder()
+      .addTime("time_day", TimeUnit.SECONDS, FieldSpec.DataType.LONG, "time_day", TimeUnit.DAYS, FieldSpec.DataType.INT)
+      .build();
+  private final Schema SCHEMA_DIFFERENT_INCOMING_OUTGOING = new Schema.SchemaBuilder()
+      .addTime("time_day", TimeUnit.SECONDS, FieldSpec.DataType.LONG, "column2", TimeUnit.DAYS, FieldSpec.DataType.INT)
+      .build();
+  private final Schema SCHEMA_NO_INCOMING = new Schema.SchemaBuilder()
```

Comment:
Same as here. If the schema has no incoming, why do we need `"incoming", TimeUnit.SECONDS, FieldSpec.DataType.LONG`?